File: 250-500/415-add-strings.cpp
string addStrings(string num1, string num2) {
    int len1 = num1.size(), len2 = num2.size();
    reverse(num1.begin(), num1.end());
    reverse(num2.begin(), num2.end());
    string ans;
    int carry = 0, sum = 0;
    char digit1 = 0, digit2 = 0;
    for (int i = 0; i < len1 || i < len2; ++i) {
        digit1 = (i < len1 ? num1[i] : '0');
        digit2 = (i < len2 ? num2[i] : '0');
        sum = (digit1 - '0') + (digit2 - '0') + carry;
        carry = (sum >= 10 ? 1 : 0);
        ans += to_string(sum % 10);
    }
    if (carry == 1) {
        ans += '1';
    }
    reverse(ans.begin(), ans.end());
    return ans;
}

File: python version/415-add-strings.py
class Solution:
    def addStrings(self, num1: str, num2: str) -> str:
        ans = ""
        num1 = num1[::-1]
        num2 = num2[::-1]
        len1 = len(num1)
        len2 = len(num2)
        carry = 0
        i = 0
        while i < len1 or i < len2:
            digit1 = int(num1[i]) if i < len1 else 0
            digit2 = int(num2[i]) if i < len2 else 0
            ans += str((digit1 + digit2 + carry) % 10)
            carry = (digit1 + digit2 + carry) // 10
            i += 1
        if carry:
            ans += "1"

        return ans[::-1]